lib/Language/Souffle/Experimental.hs

```diff
@@ -991,6 +1005,24 @@ accessorNames _ = case toStrings (Proxy :: Proxy (AccessorNames a)) of
 --   Only tuples containing up to 10 elements are currently supported.
 type Tuple ctx ts = TupleOf (MapType (Term ctx) ts)
 
+class GenVars (ts :: [Type]) where
```

Comment:
This sucks big-time. It should be an inductive definition, instead of requiring separate instances for each tuple length. Also, it should run in one of your variable number-tracking monads, but I decided that life's too short to worry about that for this illustration.